sim/midas/src/main/cc/simif.h

```diff
@@ -155,26 +172,72 @@ class simif_t {
 
   // End host-platform interface.
 
-  // LOADMEM functions
-  void read_mem(size_t addr, mpz_t &value);
-  void write_mem(size_t addr, mpz_t &value);
-  void write_mem_chunk(size_t addr, mpz_t &value, size_t bytes);
-  void zero_out_dram();
+  /// Returns the seed of the random-number generator.
+  uint64_t get_seed() const { return seed; };
 
-  uint64_t get_seed() { return seed; };
+  /// Returns the next available random number, modulo limit.
+  uint64_t rand_next(uint64_t limit) { return gen() % limit; }
 
-  // Returns the current target cycle of the fastest clock in the simulated
-  // system, based on the number of clock tokens enqueued (will report a larger
-  // number)
+  /**
+   * Returns the target cycle count.
```

Comment:
```suggestion
```
Lets not use the definite article here considering there are many target clocks at play.